### PR TITLE
Add Replicas_ListByServer to postgres flexible server 2022-03-08-preview

### PR DESCRIPTION
**Pull request by alanenriqueo**
- Add `Replicas_ListByServer` operation to postgres flexible server 2022-03-08-preview.
- Add `replicationRole` property to model `ServerPropertiesForUpdate`.
- Add `None` to `replicationRole` enum.
- Add 202 response code to `Administrators_Create`.

## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-03-08-preview/FlexibleServers.json [hunk 1]
```diff
@@ -361,6 +361,51 @@
           "nextLinkName": "nextLink"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/flexibleServers/{serverName}/replicas": {
```
Comment (ambrahma) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-03-08-preview/FlexibleServers.json:365]: Move this to a separate file Replicas.json

Comment (alanenriqueo) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-03-08-preview/FlexibleServers.json]: Done

**Review by ambrahma — APPROVED** [review]

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-03-08-preview/Administrators.json [hunk 2]
```diff
@@ -85,6 +85,9 @@
               "$ref": "#/definitions/ActiveDirectoryAdministrator"
             }
           },
+          "202": {
```
Comment (rkmanda) [inline specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-03-08-preview/Administrators.json:88]: >202 [](http://example.com/codeflow?start=11&length=3)

PUT MUST not return a 202

**Comment by rkmanda** [comment]
>           "$ref": "#/definitions/ReplicationRole",

The diffs for replicationRole changes are not clear when I do a diff between the 1st and last iterations of the PR. 




---
Refers to: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-03-08-preview/FlexibleServers.json:509 in ef004ed. [](commit_id = ef004edcd771fc275dd3a884c7a42db6ccba3ee1, deletion_comment = False)

**Comment by rkmanda** [comment]
>           "$ref": "#/definitions/ReplicationRole",

Is every file in this PR new changes? If any of them are from previous API version, please follow this: 
The first commit needs to be the previous api version and the new changes should only be added in the subsequent commits. This is explained in detail here (If using Open API which is recommended)  [https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version) or here(For manual process) :  [https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/83/Manual-Process-Adding-new-API-version](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/83/Manual-Process-Adding-new-API-version) . Please remove the "ARMChangesRequested" label once you have fixed this issue.

---
In reply to: [1309696792](https://github.com/Azure/azure-rest-api-specs/pull/21424#issuecomment-1309696792) [](http://example.com/codeflow?ancestors=1309696792)

---
Refers to: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-03-08-preview/FlexibleServers.json:509 in b9dbea9. [](commit_id = b9dbea9528f74bcf619376dda9aa0e165a5aa4fb, deletion_comment = False)

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-03-08-preview/Administrators.json (hunk 2 above)
Comment (alanenriqueo) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-03-08-preview/Administrators.json:88]: In our RP we have a lot of PUT operations that return 202. We'll fix it later, but for now we need to keep 202 in this API, and we've had this issue before: https://github.com/Azure/azure-rest-api-specs/pull/21234#pullrequestreview-1159141156

**Comment by alanenriqueo** [comment]
Hi @rkmanda, thanks for the review. This PR is only to do some fixes for the recent merged new api version 2022-03-preview (https://github.com/Azure/azure-rest-api-specs/pull/20212). We need this to be merged because CLI depends on SDK for Python and there are some urgent features needed by this month relying on this api version.

File: specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-03-08-preview/Administrators.json (hunk 2 above)
Comment (ambrahma) [reply specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2022-03-08-preview/Administrators.json:88]: @rkmanda looped you in the chat history we had earlier with Tim during earlier review regarding PUT calls returning 202.

**Review by ruowan — APPROVED** [review]

**Comment by ruowan** [comment]
This breaking change and lintDiff is existing issue. Service team already plan to fix it.

**Pull request merged**